org.eclipse.m2e.launching/src/org/eclipse/m2e/internal/launch/MavenLaunchDelegate.java

```diff
@@ -263,6 +273,24 @@ private void getPreferences(StringBuilder sb, ILaunchConfiguration configuration
       sb.append(" --threads ").append(threads);
     }
 
+    int colors = configuration.getAttribute(MavenLaunchConstants.ATTR_COLOR, 0); // Default to auto
+    if(colors == 0) { // Auto
+      // In reality we don't want to pass -Dstyle.color=auto to Maven.
+      // It tries to detect if the current stdout is a terminal (using something like `isatty`)
+      // and for Eclipse that test fails.
+      colors = isAnsiProcessingEnabled() ? 1 : 2;
```

Comment:
Same here, we should either use a constant or an ENUM!

Reply:
Done.